src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1284,5 +1284,36 @@ public void ChainWithMaxReorgPlusExtraHeadersIsCalled_AnotherChainIsPresented_Co
                 }
             }
         }
+
+        /// <summary>
+        /// Issue 31 @ Chain is 2 blocks long, CT is header 1, call PartialValidationSucceeded on header 2.
+        /// Make sure that reorg that is returned is true.
+        /// </summary>
+        [Fact]
+        public void ChainOfHeaders_CallPartialValidationSucceededOnBlockBeyondConsensusTip_ReorgIsRequired()
+        {
+            // Chain header tree setup.
+            const int initialChainSize = 1;
+            TestContext testContext = new TestContextBuilder().WithInitialChain(initialChainSize).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            ChainedHeader initialChainTip = testContext.InitialChainTip;
+
+            const int chainExtension = 1;
+            ChainedHeader chainTip = testContext.ExtendAChain(chainExtension, initialChainTip);
+            List<BlockHeader> listOfChainHeaders = testContext.ChainedHeaderToList(chainTip, 1);
+
+            // Chain is 2 blocks long: h1=h2.
+            chainedHeaderTree.ConnectNewHeaders(1, listOfChainHeaders);
+
+            // Consensus tip is h1.
+            chainedHeaderTree.ConsensusTipChanged(chainTip.Previous);
+
+            // Call PartialValidationSucceeded on h2.
+            chainedHeaderTree.PartialValidationSucceeded(chainTip, out bool reorgRequired);
```

Comment:
We did some renaming recently (not merged yet though), so it might be a good idea to get the naming consistent. The reorganisation required has been renamed here to full validation required. Could you please updated accordingly, even in the summary? Thank you